From e7d117a0e74237f7275fef29ff71b6cab5dc653b Mon Sep 17 00:00:00 2001
From: Yang Shi <yang.shi@windriver.com>
Date: Fri, 11 Nov 2011 11:25:44 -0800
Subject: [PATCH 128/238] MIPS: Octeon: Don't compile unneeded cvmx-* code into the kernel

Source: Cavium SDK 2.2-414

Without CONFIG_CAVIUM_DECODE_RSL, we don't need the CSR DB.

Also the debug code should not be included without CONFIG_CAVIUM_GDB.

This saves about 6MB of kernel size if the options are not selected.

Signed-off-by: David Daney <david.daney@cavium.com>
Integrated-by: Yang Shi <yang.shi@windriver.com>
---
 arch/mips/cavium-octeon/executive/Makefile |    6 +++---
 arch/mips/cavium-octeon/setup.c            |    2 ++
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/arch/mips/cavium-octeon/executive/Makefile b/arch/mips/cavium-octeon/executive/Makefile
index bfd95f3..a76f156 100644
--- a/arch/mips/cavium-octeon/executive/Makefile
+++ b/arch/mips/cavium-octeon/executive/Makefile
@@ -16,7 +16,7 @@ obj-y += cvmx-pko.o cvmx-spi.o cvmx-cmd-queue.o \
 	cvmx-helper-rgmii.o cvmx-helper-sgmii.o cvmx-helper-npi.o \
 	cvmx-helper-loop.o cvmx-helper-spi.o cvmx-helper-ilk.o cvmx-helper-util.o \
 	cvmx-helper-srio.o cvmx-spi4000.o cvmx-clock.o cvmx-twsi.o \
-	cvmx-srio.o cvmx-csr-db.o cvmx-csr-db-support.o cvmx-dma-engine.o cvmx-raid.o \
+	cvmx-srio.o cvmx-dma-engine.o cvmx-raid.o \
 	cvmx-ilk.o octeon-feature.o cvmx-helper-cfg.o cvmx-ipd.o
 
 obj-$(CONFIG_CAVIUM_DECODE_RSL) += cvmx-error.o cvmx-error-custom.o cvmx-error-init-cn30xx.o \
@@ -26,11 +26,11 @@ obj-$(CONFIG_CAVIUM_DECODE_RSL) += cvmx-error.o cvmx-error-custom.o cvmx-error-i
 	cvmx-error-init-cn56xx.o cvmx-error-init-cn56xxp1.o \
         cvmx-error-init-cn58xx.o cvmx-error-init-cn58xxp1.o \
         cvmx-error-init-cn63xx.o cvmx-error-init-cn63xxp1.o \
-        cvmx-error-init-cn66xx.o cvmx-error-init-cn68xx.o
+        cvmx-error-init-cn66xx.o cvmx-error-init-cn68xx.o cvmx-csr-db.o cvmx-csr-db-support.o 
 
 obj-y += cvmx-helper-errata.o cvmx-helper-jtag.o cvmx-pcie.o
 
-obj-y += cvmx-debug.o cvmx-debug-handler.o \
+obj-$(CONFIG_CAVIUM_GDB) += cvmx-debug.o cvmx-debug-handler.o \
 	 cvmx-debug-uart.o cvmx-core.o cvmx-debug-remote.o cvmx-uart.o
 
 obj-$(CONFIG_CAVIUM_OCTEON_TRA)       += cvmx-tra.o
diff --git a/arch/mips/cavium-octeon/setup.c b/arch/mips/cavium-octeon/setup.c
index 630d8a5..bf88055 100644
--- a/arch/mips/cavium-octeon/setup.c
+++ b/arch/mips/cavium-octeon/setup.c
@@ -760,7 +760,9 @@ void __init native_prom_init(void)
 #endif
 #endif
 
+#ifdef CONFIG_CAVIUM_GDB
 	cvmx_debug_init();
+#endif
 
 	octeon_setup_delays();
 
-- 
1.7.0

